JAICore/jaicore-basic/src/main/java/ai/libs/jaicore/basic/sets/PartialOrderedSet.java

```diff
@@ -278,6 +288,11 @@ public String toString() {
 	 * @return A list representing a linearization of the partial order set.
 	 */
 	public List<E> getLinearization() {
+
+		if (this.isEmpty()) {
+			return new ArrayList<>();

```

Comment:
If the user should have read access only on this anyway, we could also return here Collections.emptyList(), which is slightly better for performance. But this is only an option.